cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -744,6 +764,7 @@ private void interBrokerMoveReplicas() {
                  finishedDataMovementInMB, totalDataToMoveInMB,
                  totalDataToMoveInMB == 0 ? 100 : String.format(java.util.Locale.US, "%.2f",
                                                   (finishedDataMovementInMB * 100.0) / totalDataToMoveInMB));
+        throttleHelper.clearThrottles();
```

Comment:
Oof. Nice catch. This explains some behaviour we saw when running rebalances. OK, will look into fixing this.

Reply:
@efeg ok, I resolved this by having `waitForExecutionTaskToFinish()` return the tasks that have finished. Then I added a function to clear throttles for the tasks that have completed.

One part that's worth reviewing is that I had to figure out if a `ExecutionTask` was a valid "replication movement task", so there is a little function that calculates this. Mainly it checks that `task.brokerId() != -1` and that the state is not `IN_PROGRESS` or `STARTED`. I'm not sure how accurate this is or if there's a better way (ideally we'd have a little `TaskType` enum or class hierarchy or something to specify what kind of task each one is - if we did this then we could probably move the entire throttle code into the task itself, which is what I originally wanted to do - but this would only make sense for certain tasks).

Reply:
Thanks for the update -- I think we may avoid using `task.brokerId() != -1`. Left a comment on this.